[drone] Add RBAC support

puradox opened the pull request:
<!--
Thank you for contributing to kubernetes/charts. Before you submit this PR we'd like to
make sure you are aware of our technical requirements and best practices:

* https://github.com/kubernetes/charts/blob/master/CONTRIBUTING.md#technical-requirements
* https://github.com/kubernetes/helm/tree/master/docs/chart_best_practices

For a quick overview across what we will look at reviewing your PR, please read
our review guidelines:

* https://github.com/kubernetes/charts/blob/master/REVIEW_GUIDELINES.md

Following our best practices right from the start will accelerate the review process and
help get your PR merged quicker.

When updates to your PR are requested, please add new commits and do not squash the
history. This will make it easier to identify new changes. The PR will be squashed
anyways when it is merged. Thanks.

For fast feedback, please @-mention maintainers that are listed in the Chart.yaml file.

Please make sure you test your changes before you push them. Once pushed, a CircleCI
will run across your changes and do some initial checks and linting. These checks run
very quickly. Please check the results. We would like these checks to pass before we
even continue reviewing your changes.
-->

**What this PR does / why we need it**:
RBAC support is vital for using Drone for continuous delivery on a Kubernetes stack that is running in RBAC mode. [drone-kubernetes][2] is one such plugin that will not work on RBAC-enabled Kubernetes stack with the current version of this chart.

**Which issue this PR fixes**: fixes #4143 

**Special notes for your reviewer**:
These changes add RBAC support to Drone by creating a Service Account, Role, and Role Binding. Many of these templates were taken from the more popular [nginx-ingress][3] chart, which implements RBAC support well.

[2]: https://github.com/honestbee/drone-kubernetes
[3]: https://github.com/kubernetes/charts/tree/master/stable/nginx-ingress

puradox committed:
[drone] Add RBAC support

Implements feature request [#4143][1].

RBAC support is vital for using Drone for continuous delivery on a
Kubernetes stack that is running in RBAC mode. [drone-kubernetes][2]
is one such plugin that will not work on RBAC-enabled Kubernetes
stack with the current version of this chart.

These changes add RBAC support to Drone by creating a Service
Account, Role, and Role Binding. Many of these templates were taken
from the more popular [nginx-ingress][3] chart, which implements
RBAC well.

I hope this works for you as well as it worked for me. Cheers.

[1]: https://github.com/kubernetes/charts/issues/4143
[2]: https://github.com/honestbee/drone-kubernetes
[3]: https://github.com/kubernetes/charts/tree/master/stable/nginx-ingress

puradox committed:
Merge remote-tracking branch 'upstream/master'

* upstream/master: (32 commits)
  Add appVersion key for elasticsearch (#4721)
  Update redis/templates/NOTES.txt (#4743)
  Fix wrong image format (#4752)
  Add appVersion for cassandra (#4719)
  Add appVersion key for factorio (#4718)
  Add appVersion key for gcloud-sqlproxy (#4716)
  Add appVersion key for heapster (#4715)
  Add appVersion for kube-ops-view (#4705)
  Add home and appVersion key for minecraft (#4694)
  Typo fix in webpagetest-agent/README.md (#4579)
  [stable/nginx-ingress] Add support for extraContainers, extraVolumeMounts and extraVolumes (#3983)
  Add appVersion key for etcd (#4722)
  Add appVersion for kube-registry-proxy (#4724)
  Add appVersion key for mysqlha (#4725)
  Add home and appVersion for redis-cache (#4726)
  Add appVersion for webpagetest-server (#4728)
  Fix missing ghost.fullname in template call. (#4611)
  [chartmuseum] move from incubator to stable (#4601)
  Bump cockroachdb chart to version 2.0.0 (#4745)
  Update Concourse to 3.10 (#4542)
  ...

puradox committed:
[drone] Fix ingress path for classes other than gce

A change was introduced in #3949 that incorrectly used the "and"
template function. It was used infix (B and C) rather than prefix
(and B C). This led to a subtle bug where it using the path for gce
ingress for all ingress classes (e.g. nginx).

Additionally, the "and" template function does not support short
circuit evaluation, so two if statements had to be introduced. This
made for some mess, but it was neccessary.

timstoop commented on incubator/drone/values.yaml:
```diff
@@ -178,3 +178,12 @@ persistence:
 # sharedSecret: supersecret
+
+## Install Default RBAC roles and bindings
+rbac:
```
Can you add these to the README as well, please?

puradox replied:
Nice catch. I'll add those entries in right now

timstoop commented:
/ok-to-test

puradox committed:
[drone] Removed unused value - rbac.roleRef

puradox committed:
[drone] Add docs for rbac values

timstoop commented:
/lgtm

puradox commented:
/assign @mattfarina

unguiculus commented:
Please follow our RBAC best practices:
https://github.com/kubernetes/helm/blob/master/docs/chart_best_practices/rbac.md

/hold
/assign

puradox committed:
[drone] Use RBAC best practices

https://github.com/kubernetes/helm/blob/master/docs/chart_best_practices/rbac.md

This also changes the default RBAC api version to v1
from v1beta1.

puradox committed:
Merge remote-tracking branch 'upstream/master'

* upstream/master: (38 commits)
  [stable/fluent-bit] Cater for full set of Elastic Output Plugin parameters (Auth & TLS) (#4571)
  [drone] Fix ingress path for nginx-ingress (#4771)
  [stable/rabbitmq-ha] Add ingress for management and customizable probes (#4749)
  update readme of traefik (#4786)
  Update version! (#4783)
  Readme fix: default version is 0.12.0 (#4707)
  adding redis-exporter chart (#2743)
  [stable/rabbitmq-ha] Add option to use existing certificate (#4114)
  [stable/rabbitmq-ha] Clustering fixes (#4610)
  [stable/redmine] Release 3.0.1 (#4776)
  update selenium readme (#4775)
  Add MSSQL Master DB persistence (#4701)
  [dex] - initial chart for coreos/dex (#4580)
  [stable/rabbitmq-ha] Fix NOTES (#4681)
  [stable][selenium] Add the option of adding nodeSelectors (#3791)
  Add home key in yaml and typo fix in burrow/README.md (#4558)
  Add appVersion key for check-mk and fix Yaml format error (#4754)
  Add home and appVersion keys for sentry-kubernetes (#4755)
  fix zookeeper supported kubernetes version (#4774)
  fix typo in opencart (#4751)
  ...

unguiculus commented on incubator/drone/templates/role-binding.yaml:
```diff
@@ -0,0 +1,19 @@
+metadata:
+  name: {{ template "drone.fullname" . }}
+  labels:
+    app: {{ template "drone.fullname" . }}
```
`app: {{ template "drone.name" . }}`

unguiculus commented on incubator/drone/templates/role.yaml:
```diff
@@ -0,0 +1,21 @@
+metadata:
+  name: {{ template "drone.fullname" . }}
+  labels:
+    app: {{ template "drone.fullname" . }}
```
`app: {{ template "drone.name" . }}`

unguiculus commented on incubator/drone/templates/service-account.yaml:
```diff
@@ -0,0 +1,11 @@
+metadata:
+  name: {{ template "drone.serviceAccountName" . }}
+  labels:
+    app: {{ template "drone.fullname" . }}
```
`app: {{ template "drone.name" . }}`

puradox replied:
I have changed the app label to match 👍

puradox committed:
[drone] Fix rbac "app" label

Every other template is setting the app label to
drone.name, so the rbac resources should match.

Thank @unguiculus for catching this

unguiculus commented:
/lgtm

timstoop commented:
@unguiculus can this continue or is it waiting on another fix before this one gets applied?

unguiculus commented:
/hold cancel
